06-events-practice/2-tooltip/index.js

```diff
@@ -1,6 +1,65 @@
 class Tooltip {
-  initialize () {
+  static instance = null;
 
+  constructor() {
+    if (Tooltip.instance) {
+      return Tooltip.instance; 
+    }
+
+    Tooltip.instance = this;
+
+    this.element = null; 
+    this.targetElement = null; 
+  }
+
+  initialize() {
+    const container = document.createElement('div');
+    container.innerHTML = `<div id="container" data-tooltip="bar-bar-bar">
+      Aperiam consectetur dignissimos dolores ex mollitia.
+    </div>`;
+
+    this.targetElement = container.firstElementChild; 
+    this.element = document.createElement('div'); 
+
+    document.body.append(container); 
+    document.body.append(this.element); 
+
+    this.addEventListeners();
+  }
+
+  addEventListeners() {
+    this.targetElement.addEventListener('pointerover', this.showTooltip.bind(this));
```

Comment:
this.targetElement.addEventListener('pointerover', this.handlePointeroverTargetElement);